py3status/modules/external_script.py

```diff
@@ -51,11 +57,13 @@ class Py3status:
     # available configuration parameters
     cache_timeout = 15
     format = '{output}'
+    format_notification = None
```

Comment:
`button_refresh` is used because some modules connects to Internet. We don't want people to scroll and start hitting up Internet.

It's in `external_script`, `whatismyip`, `github`, `google_calendar`.

Reply:
Talking specifically about mouse scroll wheel. We can get 50+ events per second if we scroll hard.... I don't think it's good idea to force notifications on existing `external_script` users because we don't know their cases and most of them can not want it... so they have to fix thier config to turn it off... Having it off (and always off by default) is better.

Reply:
Well not true, it's just hiding the issue - those who enabled notifications will suffer from this issue. Lets introduce button_show_notification then

Reply:
Why do we need `button_show_notification` if it works okay on several buttons?

Reply:
Because like you said, I can accidentally generate 50+ events with mouse scroll - your argument "I don't want such accidents to happen, therefore disable notifications by default", my argument "I _also_ don't want this to happen, but disabling notifications is not an option for me, so let's find a better way to fix this issue".

Reply:
I did not disable notifications because of 50+ events. When we add `on_click` to the module, the automagically `2` is turned off. For that reason alone, we sometimes need to add it to deal with Internet modules. For other modules, we just let it refresh.

In this case, `button_refresh` is not added because there simply are no other button. Your use case is that you wanted the notification on last output and at same not, not triggering anything.

I disable notifications because it is simply a bad behavior to force notifications on other users.

Reply:
Obviously we don't agree on same things.... :-) 

Pinging @ultrabug for his opinion on this matter.

Reply:
I guess I simply don't understand your use-case, with what you are proposing most buttons will not be doing anything (not even module refresh, absolutely nothing). Why not put some useful action on button click if these buttons are unused? 

By analogy we can say that Chromium is forcing an action "open new window" when you press "Ctrl+N". This shortcut is unused, so they put some action on it. Same here, button is unused and I have a useful action that I can put on this button.

I'll update the code in this PR to have it show my intention for this, and yeah let's have a third opinion on it 🙂

Reply:
Because this is `external_script`. It does not really have a specific purpose and your use case is a bit different in that you only wanted a notification on last output... External scripts often have some intervals in them... and `format_notification` might confuse users in thinking that they'll be getting a notification on every interval... (say I put in 7200, then I expect a notification every two hours). This is not the case... so I'm not sure if we should even add `format_notification` here too... It's too early to tell.

What you wanted is minimally implemented.  Minimal is the keyword here.  If users come by wanting notification on every two hours, then yes, it would be easier to implement that minimally too... 

If we add buttons, then it's hard to undo them so I'm trying to avoid not adding unnecessary configs because if an user stop by with a PR containing something really useful for `external_script` and it requires a button, then by all means, we should let him do that.

`external_script` is hard in that it does not really have specific purposes... It might be better to enhance https://py3status.readthedocs.io/en/latest/configuration.html#special-on-click-commands to accept `{placeholders}` too, not just `$OUTPUT`...

And obviously, to enhance `on_click` too with new options to better address #1264.

Reply:
> if an user stop by with a PR containing something really useful for external_script and it requires a button, then by all means, we should let him do that.

ah, my feelings are hurt 😢, my PR is not considered cool/useful enough to make a case for reserving a button... 😛 

Although I find it strange that we prefer to let notifications be shown on mouse scroll to creating a dedicated `button_show_notification`, I'm fine with conceding this to you — in reality there is a throttle on notifications, so even if you scroll extensively, you will get at most one notification every 5 seconds. Accidents will not be too painful.

However, I still believe that by default showing a notification on click is better than doing absolutely nothing. If you are annoyed by notification, no configuration change is needed — simply don't click on the module, click is not doing anything except showing a notification.

Reply:
>However, I still believe that by default showing a notification on click is better than doing absolutely nothing.

Doing absolutely nothing is the best way to ensure that we don't break user configs. Instead of trying to put something in the `format_notification` by default... only to later remove it., we break one thing. We can defer the `format_notification` to users instead.

This is `external_script` though. Users uses this to execute their scripts. We don't know their scripts. Their scripts could already contain some notifications too... so now they get dual notifications... One for clicking the module (and not caring about last output)... and one for when  when their script is done doing something.

I believe there can be some grief to have notifications enabled by default versus none.

_Still waiting for third opinion... No rush... Toy with the current module._